File: ai-service/config/env.py
import os
import sys

REQUIRED_VARS = ["GEMINI_API_KEY", "AI_SERVICE_SECRET"]

def validate_env():
    missing = [key for key in REQUIRED_VARS if not os.getenv(key)]
    if missing:
        print(f"Missing required environment variables: {', '.join(missing)}")
        sys.exit(1)
    print("All required environment variables are set")

File: backend/src/config/env.js
// Optional: Validate environment variables
const requiredEnvVars = [
  "DATABASE_URL",
  "JWT_SECRET",
  "JWT_EXPIRE",
  "JWT_REFRESH_SECRET",
  "JWT_REFRESH_EXPIRE",
  "FRONTEND_URL",
  "GOOGLE_CLIENT_ID",
  "GOOGLE_CLIENT_SECRET",
  "GOOGLE_CALLBACK_URL",
];

const validateEnv = () => {
  const missing = requiredEnvVars.filter((envVar) => !process.env[envVar]);

  if (missing.length > 0) {
    console.error(
      `❌ Missing required environment variables: ${missing.join(", ")}`,
    );
    process.exit(1);
  }

  console.log("✅ All required environment variables are set");
};

module.exports = { validateEnv };
